src/core/runtime.cpp

```diff
@@ -3139,4 +3072,21 @@ SHCore *__cdecl shardsInterface(uint32_t abi_version) {
 
   return result;
 }
-};
+}
+
+namespace shards {
+void decRef(ShardPtr shard) {
+  assert(shard->refCount > 0);
+  --shard->refCount;
+  if (shard->refCount == 0) {
+    SHLOG_TRACE("DecRef 0 shard {:x} {}", (size_t)shard, shard->name(shard));
+    shard->destroy(shard);
+  }
+}
+
+void incRef(ShardPtr shard) {
+  if (shard->refCount == 0)
+    SHLOG_TRACE("IncRef 0 shard {:x} {}", (size_t)shard, shard->name(shard));
+  ++shard->refCount;
```

Comment:
Shouldn't this be atomic? Or do we guarantee that ShardRef are never shared between threads?

Reply:
Hmm yeah good point, it should be